autoload/coverage/lcov/parsing.vim

```diff
@@ -0,0 +1,130 @@
+" Copyright 2021 Google Inc. All rights reserved.
+"
+" Licensed under the Apache License, Version 2.0 (the "License");
+" you may not use this file except in compliance with the License.
+" You may obtain a copy of the License at
+"
+"     http://www.apache.org/licenses/LICENSE-2.0
+"
+" Unless required by applicable law or agreed to in writing, software
+" distributed under the License is distributed on an "AS IS" BASIS,
+" WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+" See the License for the specific language governing permissions and
+" limitations under the License.
+
+"{{{ Init
+
+let s:plugin = maktaba#plugin#Get('coverage')
+
+"}}}
+
+"{{{ [lcov](https://github.com/linux-test-project/lcov) coverage provider.
+
+""
+" @private
+" Categorizes a line potentially containing coverage information in a coverage
+" file.  Recognized coverage information is be formatted as:
+" - DA:<line number>,<execution count>[,<checksum>]
+" - BA:<line number>,<branch coverage (0: uncovered, 1: partial, 2: covered>
+" - BRDA:<line number>,<block number>,<branch number>,<taken>
+"
+" For BRDA, we can't get partial coverage (only '-' for uncovered, or 1+)
+function! s:TryParseLine(line) abort
+  if maktaba#string#StartsWith(a:line, 'BA:') ||
+        \ maktaba#string#StartsWith(a:line, 'DA:')
+    let l:hits_index = 1
+  elseif maktaba#string#StartsWith(a:line, 'BRDA:')
+    let l:hits_index = 3
+  else
+    return []
+  endif
+
+  try
+    let [l:prefix, l:raw_info] = split(a:line, ':')
+    let l:info = split(l:raw_info, ',')
+    let l:linenum = str2nr(info[0])
+    let l:hits = str2nr(info[l:hits_index])  " Will return 0 if hits is '-'
```

Comment:
I don't understand this note about `'-'`. Could you clarify? (Seems a little magical.)

Reply:
Expanded a bit in the code.  lcov expresses "uncovered" BRDA as '-' (instead of '0'), so I was expecting to have to explicitly handle the '-' case, here.

I added this comment because instead, `str2nr` _was_ being a bit magical by defaulting non-numerical strings to 0.  Unrelated to this particular instance, but it also appears to strip whitespace (both behaviors are undocumented).

Reply:
You're right, it's one of more bizarre vimscript features. Pretty clear now with your updated comment, though. Thanks!